### PR TITLE
Silence QueryRunner startup logs in tests

### PR DESCRIPTION
**Pull request by nineinchnick**
Silence QueryRunner startup logs to make it easier to browse logs in case of a failure. For example, the output of the `TestCachingHiveMetastoreWithQueryRunner` test from `trino-hive-hadoop2` module goes down from 2348 to 172 lines. Silenced info should be repetitive across different CI runs and not convey information specific to tests. This includes non-default server configuration properties, which are passed explicitly in the QueryRunner builder.

Silenced logger examples:
`org.glassfish.jersey.internal.inject.Providers` and `org.glassfish.jersey.internal.Errors`:
```
2022-01-19T07:36:49.107-0600 WARNING A provider io.airlift.discovery.server.ServiceResource registered in SERVER runtime does not implement any provider interfaces applicable in the SERVER runtime. Due to constraint configuration problems the provider io.airlift.discovery.server.ServiceResource will be ignored. 
```

`org.eclipse.jetty.server`:
```
2022-01-19T07:36:49.233-0600 INFO Started o.e.j.s.ServletContextHandler@5e0edcd1{/,null,AVAILABLE,@http}
2022-01-19T07:36:49.240-0600 INFO Started http@106b1b81{HTTP/1.1, (http/1.1, h2c)}{127.0.0.1:56868}
2022-01-19T07:36:49.241-0600 INFO Started @5157ms
```

`io.airlift.bootstrap`, `io.trino.server.PluginManager`:
```
2022-01-19T07:36:49.345-0600 INFO Life cycle starting...
2022-01-19T07:36:49.346-0600 INFO Life cycle started
2022-01-19T07:36:53.748-0600 INFO Registering connector tpch
2022-01-19T07:36:53.750-0600 INFO Registering connector tpch
2022-01-19T07:36:53.751-0600 INFO Registering connector tpch
2022-01-19T07:36:53.787-0600 INFO Registering connector clickhouse
2022-01-19T07:36:53.787-0600 INFO Registering connector clickhouse
2022-01-19T07:36:53.788-0600 INFO Registering connector clickhouse
```

`Bootstrap`:
```
2022-01-19T07:36:53.862-0600 INFO PROPERTY                                                   DEFAULT      RUNTIME                             DESCRIPTION
2022-01-19T07:36:53.863-0600 INFO clickhouse.map-string-as-varchar                           false        true                                Map ClickHouse String and FixedString as varchar instead of varbinary
```

Previous failed attempts to reduce amount of logs:
* #10592 - closed because it would require to download artifacts with logs
* #10620 - closed because we run tests in parallel

### CODE REVIEW

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java [hunk 1]
```diff
@@ -46,6 +45,7 @@ public static QueryRunner createSqlServerQueryRunner(
             Iterable<TpchTable<?>> tables)
             throws Exception
     {
+        DistributedQueryRunner.setupLogging();
```
Comment (ebyhr) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:48]: Bootstrap log is helpful in SQL Server connector when connecting to the instance from other tool (e.g. DBeaver) during the development because the connector generates the random database name in `TestingSqlServer`.

Comment (nineinchnick) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:48]: Do you think we could log this explicitly in the QueryRunner instead of relying on the Bootstrap logger? The latter includes lots of irrelevant entries.

Comment (nineinchnick) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:48]: In case of SQLServer test container there's a constant password, but if it were random, you wouldn't get it from the Bootstral logger anyway, so explicit logging might be better.

Comment (nineinchnick) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:48]: Something like this:
```
2022-01-20T03:49:50.549-0600	INFO	pool-4-thread-1	io.trino.testing.DistributedQueryRunner	Created catalog sqlserver (sqlserver) in 421.28ms, with properties: {connection-password=A_Str0ng_Required_Password, connection-url=jdbc:sqlserver://localhost:62774;database=database_3a61c50f7ce84546b06bb56e2c313219;lockTimeout=60000, connection-user=sa, allow-drop-table=true}
```

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 2]
```diff
@@ -432,7 +447,7 @@ public void createCatalog(String catalogName, String connectorName, Map<String,
             catalogNames.add(server.createCatalog(catalogName, connectorName, properties));
         }
         CatalogName catalog = getOnlyElement(catalogNames);
-        log.info("Created catalog %s (%s) in %s", catalogName, catalog, nanosSince(start));
+        log.info("Created catalog %s (%s) in %s, with properties: %s", catalogName, catalog, nanosSince(start), properties);
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:450]: Logging properties in `DistributedQueryRunner` is dangerous for SaaS connector. We should consider other way. e.g. Move logging place to other places, prepare toggle and disable by default and so on.

Comment (nineinchnick) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java]: Thanks, that would be a serious issue. I moved this to the SqlServerQueryRunner.

Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java]: We should log properties via Bootstrap since it's aware of ConfigSecuritySensitive and can mask logs properly. Anything else seems error prone.

Comment (nineinchnick) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java]: We're talking about tests here only. I reverted this change in the `DistributedQueryRunner`.

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java [hunk 3]
```diff
@@ -58,6 +58,7 @@ public static DistributedQueryRunner createH2QueryRunner(Iterable<TpchTable<?>>
     public static DistributedQueryRunner createH2QueryRunner(Iterable<TpchTable<?>> tables, Map<String, String> properties, Module module)
             throws Exception
     {
+        DistributedQueryRunner.setupLogging();
```
Comment (kokosing) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java:61]: Can we make this part of `DistributedQueryRunner.Builder`, so your config would be default while user could reset it to plain `Logging.initialize()` or maybe configure in a custom way?

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 4]
```diff
@@ -97,6 +100,18 @@
...
 
+    public static Logging setupLogging()
+    {
+        Logging logging = Logging.initialize();
+        logging.setLevel("Bootstrap", WARN);
+        logging.setLevel("io.airlift.bootstrap", WARN);
+        logging.setLevel("org.glassfish.jersey.internal.inject.Providers", ERROR);
+        logging.setLevel("org.glassfish.jersey.internal.Errors", ERROR);
+        logging.setLevel("org.eclipse.jetty.server", WARN);
+        logging.setLevel("io.trino.server.PluginManager", WARN);
```
Comment (kokosing) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:111]: I would prefer to keep `io.trino` and `io.airlift` on `INFO`.  And then we can see what to do next.

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 5]
```diff
@@ -97,6 +100,18 @@
         return new Builder<>(defaultSession);
     }
 
+    public static Logging setupLogging()
+    {
+        Logging logging = Logging.initialize();
+        logging.setLevel("Bootstrap", WARN);
+        logging.setLevel("io.airlift.bootstrap", WARN);
+        logging.setLevel("org.glassfish.jersey.internal.inject.Providers", ERROR);
+        logging.setLevel("org.glassfish.jersey.internal.Errors", ERROR);
```
Comment (kokosing) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:109]: Why not `org.glassfish=ERROR`?

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java (hunk 3 above)
Comment (nineinchnick) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java:61]: The reason I did it outside is that in some test classes logging is set up once in a `@BeforeClass` method and there are multiple query runners created in tests. The loggers for which this configures levels are not tied to a particular query runner, so it might be surprising they keep changing after creating new query runners.

File: plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java [hunk 6]
```diff
@@ -48,6 +47,7 @@ private AccumuloQueryRunner() {}
     public static synchronized DistributedQueryRunner createAccumuloQueryRunner(Map<String, String> extraProperties)
             throws Exception
     {
+        DistributedQueryRunner.setupLogging();
```
[the remark this answers is not in the document]
Comment (nineinchnick) [reply plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java:50]: What do you mean by "test gets instantiated"? This is called in `@BeforeClass` or `@BeforeMethod`, even when a query runner would get created in the test itself.

If anyone forgets to call this nothing bad will happen and we can always add it back later.

[the remark this answers is not in the document]
Comment (findepi) [reply plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java:50]: > What do you mean by "test gets instantiated"?

i meant initialized, sorry

[the remark this answers is not in the document]
Comment (electrum) [reply plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java:50]: I agree, we can simply call this inside the build method for `DistributedQueryRunner`. No need to touch every single query runner.

File: plugin/trino-elasticsearch/pom.xml [hunk 7]
```diff
@@ -18,6 +18,16 @@
         <dep.elasticsearch.version>6.0.0</dep.elasticsearch.version>
     </properties>
 
+    <dependencyManagement>
```
Comment (electrum) [inline plugin/trino-elasticsearch/pom.xml:21]: Why? Seems unrelated

File: testing/trino-testing/pom.xml [hunk 8]
```diff
@@ -16,6 +16,16 @@
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
     </properties>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.glassfish.jersey.core</groupId>
+                <artifactId>jersey-common</artifactId>
+                <version>2.26</version>
```
Comment (electrum) [inline testing/trino-testing/pom.xml:24]: Don't add the dependency here. It's not directly used by this module. You can work around the dependency plugin false positive by building the string via concatenation, or by using `<ignoredUsedUndeclaredDependencies>` for the plugin.

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 9]
```diff
@@ -97,6 +100,18 @@
...
+        logging.setLevel("org.eclipse.jetty.server", WARN);
+        logging.setLevel("io.trino.server.PluginManager", WARN);
+        return logging;
```
Comment (electrum) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:112]: No need to return this. `Logging.initialize()` returns the singleton. If anyone needs it they can just ask for it.

File: testing/trino-tests/src/test/java/io/trino/execution/TestConnectorEventListener.java [hunk 10]
```diff
@@ -47,6 +47,7 @@ public void setUp()
                 .setSchema("tiny")
                 .setClientInfo("{\"clientVersion\":\"testVersion\"}")
                 .build();
+        DistributedQueryRunner.setupLogging();
```
Comment (electrum) [inline testing/trino-tests/src/test/java/io/trino/execution/TestConnectorEventListener.java:50]: Do this in the constructor instead of putting this in all the tests.

**Review by electrum — CHANGES_REQUESTED** [review]

File: testing/trino-testing/pom.xml (hunk 8 above)
Comment (nineinchnick) [reply testing/trino-testing/pom.xml]: Thanks, I didn't know there can be false positives. I rebased on master now, removed these entries and the dependency plugin doesn't complain anymore, weird.

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java (hunk 3 above)
Comment (nineinchnick) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/H2QueryRunner.java]: Moved to the constructor.

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java (hunk 4 above)
Comment (nineinchnick) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java]: Done!

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java (hunk 5 above)
Comment (nineinchnick) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java]: Done!

File: plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (nineinchnick) [reply plugin/trino-accumulo/src/test/java/io/trino/plugin/accumulo/AccumuloQueryRunner.java]: Done!

File: plugin/trino-elasticsearch/pom.xml (hunk 7 above)
Comment (nineinchnick) [reply plugin/trino-elasticsearch/pom.xml]: Removed.

File: testing/trino-tests/src/test/java/io/trino/execution/TestConnectorEventListener.java (hunk 10 above)
Comment (nineinchnick) [reply testing/trino-tests/src/test/java/io/trino/execution/TestConnectorEventListener.java]: Done!

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java [hunk 11]
```diff
@@ -61,6 +62,7 @@ public static QueryRunner createSqlServerQueryRunner(
 
             queryRunner.installPlugin(new SqlServerPlugin());
             queryRunner.createCatalog(CATALOG, "sqlserver", connectorProperties);
+            log.info("%s catalog properties: %s", CATALOG, connectorProperties);
```
Comment (kokosing) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:65]: It is not needed. It is already logged.

**Review by kokosing — COMMENTED** [review]
Looks good. What improvement did we get?

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java (hunk 11 above)
Comment (nineinchnick) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/SqlServerQueryRunner.java:65]: This PR disables the regular `Bootstrap` logger that would log this. This has been added since the DB name is random. It gets logged as:
```
 2022-01-27T11:23:16.813-0600	INFO	pool-4-thread-1	io.trino.plugin.sqlserver.SqlServerQueryRunner	sqlserver catalog properties: {connection-*** connection-url=jdbc:sqlserver://localhost:49165;database=database_e6cbbe8278d54bd791b508e1ef187d3d;lockTimeout=60000, connection-user=sa, allow-drop-table=true}
```
You can see that there are no other occurrences of the db name in the CI log, except for failed query error messages.

**Comment by nineinchnick** [comment]
> Looks good. What improvement did we get?

See the PR description - some jobs go down from thousands to hundreds of log lines.

**Comment by nineinchnick** [comment]
@electrum your requested changes have been applied, PTAL

**Review by electrum — APPROVED** [review]

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java [hunk 12]
```diff
@@ -128,8 +127,8 @@ public static Session createSession()
     public static void main(String[] args)
             throws Exception
     {
-        Logging.initialize();
-
+        // To start Elasticsearch:
+        // docker run -p 9200:9200 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.6.2
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java:131]: We can simply remove this line instead of a separate commit.

@nineinchnick By the way, I added `ElasticsearchExternalQueryRunner` few days ago. Could you remove the logging initialization from the class if necessary?

[the remark this answers is not in the document]
Comment (nineinchnick) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchQueryRunner.java]: This is a bad rebase, sorry, I should pay more attention. Thanks for spotting this. I applied the changes in the other runner.
